agent/agent.go

```diff
@@ -21,13 +21,41 @@ import (
 	"github.com/google/gops/signal"
 )
 
-// Start starts the gops agent on a host process. Once agent started,
-// users can use the advanced gops features.
+// Agent represents an agent that enable the advanced gops features
+type Agent struct {
```

Comment:
I was thinking of a global Listen and global Close without a handle to the agent.

```
func Listen(opts *AgentOptions) error
func Close() error
```